### PR TITLE
I need "switch to the named window"

### PR DESCRIPTION
**Pull request by prettyhappycatty**
I need "switch to the named window”.

Where should i make its test code?

### CODE REVIEW

**Comment by NA070** [comment]
Thank you for your PR!

Please add test here.
https://github.com/lifull-dev/bucky-core/blob/master/spec/test_equipment/user_operation/user_operation_helper_spec.rb

**Comment by prettyhappycatty** [comment]
Thank you for your comment about where test code is.

I've used RSpec first time :)
( i've used selenium with Java, and i've used ruby little bit)

> Thank you for your PR!
> 
> Please add test here.
> https://github.com/lifull-dev/bucky-core/blob/master/spec/test_equipment/user_operation/user_operation_helper_spec.rb

File: lib/bucky/test_equipment/user_operation/user_operation_helper.rb [hunk 1]
```diff
@@ -49,6 +49,11 @@ def back_to_window(_)
           @driver.switch_to.window(@driver.window_handles.first)
         end
 
+        def switch_to_the_window(args)
+          sleep 1
```
Comment (NA070) [inline lib/bucky/test_equipment/user_operation/user_operation_helper.rb]: There is a question I'd like to ask.
Why "sleep" need here?

Comment (prettyhappycatty) [reply lib/bucky/test_equipment/user_operation/user_operation_helper.rb]: Hi, thank you for the question.

it's to wait opening another window.
But, Is it better to write "click_and_wait" function?

Comment (NA070) [reply lib/bucky/test_equipment/user_operation/user_operation_helper.rb]: Thanks for your answer.

> Is it better to write "click_and_wait" function?

No. 
I think switch_to_the_window should have wait processing.
But, 1 sec is too long.

How about the following implementation?
```
def switch_to_the_window(args)
  counter = 10
  begin
    counter -= 1
    @driver.switch_to.window(args[:window_name])
  rescue Selenium::WebDriver::Error::NoSuchWindowError
    if counter > 0
      sleep 0.1
      retry
    end
    raise Selenium::WebDriver::Error::NoSuchWindowError, "Exceeded the limit for trying shitch_to_the_window.\n    #{$ERROR_INFO.message}"
  end
end
```

Comment (prettyhappycatty) [reply lib/bucky/test_equipment/user_operation/user_operation_helper.rb]: that sounds so nice!
mine is a waste of time and yours isn't!

i remember now about webdriver.wait func.
How about to use "webdriver.wait(hasTheWindowAlreadyAppeard)".

i don't know it on ruby, i'll look into it.

Comment (prettyhappycatty) [reply lib/bucky/test_equipment/user_operation/user_operation_helper.rb]: I wrote with webdriver.wait method.
How about this?

```
def switch_to_the_window(args)
  #change ignore condition, NoSuchElementError to NoSuchWindowError
  wait = Selenium::WebDriver::Wait.new(:timeout => 1, :interval => 0.1, :ignore => [Selenium::WebDriver::Error::NoSuchWindowError])
  begin
    # when successfully switched, return of switch_to.window is nil.
    wait.until{ @driver.switch_to.window(args[:window_name]).nil? }
  rescue Selenium::WebDriver::Error::TimeoutError
    raise Selenium::WebDriver::Error::TimeoutError, "Exceeded the limit for trying shitch_to_the_window.\n "
  end
end
```

Comment (NA070) [reply lib/bucky/test_equipment/user_operation/user_operation_helper.rb]: Cool!
Please implement it!

Comment (prettyhappycatty) [reply lib/bucky/test_equipment/user_operation/user_operation_helper.rb]: thanks!

i implemented.
Ummm....   < this pull request is 66.6% (50% is the threshold).

Comment (NA070) [reply lib/bucky/test_equipment/user_operation/user_operation_helper.rb]: It is no problem!
I checked in my PC.
COVERAGE:  96.72% -- 59/61 lines in 1 files

```
+----------+------------------------------------------------------------------+-------+--------+---------+
| coverage | file                                                             | lines | missed | missing |
+----------+------------------------------------------------------------------+-------+--------+---------+
|  96.72%  | lib/bucky/test_equipment/user_operation/user_operation_helper.rb | 61    | 2      | 58-59   |
+----------+------------------------------------------------------------------+-------+--------+---------+
```

File: spec/test_equipment/user_operation/user_operation_helper_spec.rb [hunk 2]
```diff
@@ -58,6 +58,15 @@
     end
   end
 
+  describe '#switch_to_the_window' do
```
Comment (NA070) [inline spec/test_equipment/user_operation/user_operation_helper_spec.rb]: Could you implement test about raising timeout error?

Comment (prettyhappycatty) [reply spec/test_equipment/user_operation/user_operation_helper_spec.rb]: maybe...
i'll try it!

Comment (NA070) [reply spec/test_equipment/user_operation/user_operation_helper_spec.rb]: Implementation example:
https://github.com/lifull-dev/bucky-core/blob/master/spec/test_equipment/user_operation/user_operation_helper_spec.rb#L137

Comment (prettyhappycatty) [reply spec/test_equipment/user_operation/user_operation_helper_spec.rb]: thanks!
I see and know how to check TimeOutException.

But how to raise TimeOutExeption is bothering me.

**Pull request merged**